File: app/src/main/java/com/elementary/tasks/core/dialogs/VoiceResultDialog.java
package com.elementary.tasks.core.dialogs;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;

import com.elementary.tasks.R;
import com.elementary.tasks.core.data.models.Reminder;
import com.elementary.tasks.core.utils.Constants;
import com.elementary.tasks.core.utils.Dialogues;
import com.elementary.tasks.core.view_models.reminders.ReminderViewModel;
import com.elementary.tasks.databinding.ReminderListItemBinding;
import com.elementary.tasks.reminder.create_edit.CreateReminderActivity;

import androidx.lifecycle.ViewModelProviders;

/**
 * Copyright 2017 Nazar Suhovich
 * <p/>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p/>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p/>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

public class VoiceResultDialog extends BaseDialog {

    private DialogInterface.OnCancelListener mCancelListener = dialogInterface -> finish();
    private DialogInterface.OnDismissListener mOnDismissListener = dialogInterface -> finish();

    @Override
    protected final void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        int id = getIntent().getIntExtra(Constants.INTENT_ID, 0);

        ReminderViewModel viewModel = ViewModelProviders.of(this, new ReminderViewModel.Factory(getApplication(), id)).get(ReminderViewModel.class);
        viewModel.reminder.observe(this, reminder -> {
            if (reminder != null) {
                showReminder(reminder);
            }
        });
    }

    private void showReminder(Reminder reminder) {
        AlertDialog.Builder alert = Dialogues.getDialog(this);
        alert.setTitle(getString(R.string.saved));

        ReminderListItemBinding binding = ReminderListItemBinding.inflate(LayoutInflater.from(this), null, false);
        binding.setItem(reminder);
        binding.itemCheck.setVisibility(View.GONE);
        binding.reminderContainer.setBackgroundColor(getThemeUtil().getCardStyle());
        alert.setView(binding.getRoot());
        alert.setCancelable(true);
        alert.setNegativeButton(R.string.edit, (dialogInterface, i) -> {
            dialogInterface.dismiss();
            startActivity(new Intent(VoiceResultDialog.this, CreateReminderActivity.class).putExtra(Constants.INTENT_ID, reminder.getUniqueId()));
            finish();
        });
        alert.setPositiveButton(R.string.ok, (dialog, id) -> {
            dialog.dismiss();
            finish();
        });
        AlertDialog alertDialog = alert.create();
        alertDialog.setOnCancelListener(mCancelListener);
        alertDialog.setOnDismissListener(mOnDismissListener);
        alertDialog.show();
    }
}

File: app/src/main/java/com/elementary/tasks/core/dialogs/VoiceResultDialog.kt
package com.elementary.tasks.core.dialogs

import android.os.Bundle
import android.widget.LinearLayout
import com.elementary.tasks.R
import com.elementary.tasks.core.data.ui.UiReminderList
import com.elementary.tasks.core.data.ui.UiReminderListActive
import com.elementary.tasks.core.utils.Constants
import com.elementary.tasks.core.utils.ui.Dialogues
import com.elementary.tasks.reminder.ReminderBuilderLauncher
import com.elementary.tasks.reminder.lists.adapter.ReminderViewHolder
import com.elementary.tasks.voice.VoiceResultDialogViewModel
import org.koin.android.ext.android.inject
import org.koin.androidx.viewmodel.ext.android.viewModel
import org.koin.core.parameter.parametersOf

class VoiceResultDialog : BaseDialog() {

  private val viewModel by viewModel<VoiceResultDialogViewModel> { parametersOf(getId()) }
  private val reminderBuilderLauncher by inject<ReminderBuilderLauncher>()

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    viewModel.reminder.observe(this) { showReminder(it) }
    lifecycle.addObserver(viewModel)
  }

  private fun getId() = intent.getStringExtra(Constants.INTENT_ID) ?: ""

  private fun showReminder(reminder: UiReminderList) {
    val alert = dialogues.getMaterialDialog(this)
    alert.setTitle(getString(R.string.saved))

    val parent = LinearLayout(this)
    parent.layoutParams = LinearLayout.LayoutParams(
      LinearLayout.LayoutParams.MATCH_PARENT,
      LinearLayout.LayoutParams.WRAP_CONTENT
    )
    parent.orientation = LinearLayout.VERTICAL

    val holder = ReminderViewHolder(parent, editable = false)
    holder.setData(reminder as UiReminderListActive)

    parent.addView(holder.itemView)

    alert.setView(parent)
    alert.setCancelable(true)
    alert.setNegativeButton(R.string.edit) { dialogInterface, _ ->
      dialogInterface.dismiss()
      reminderBuilderLauncher.openLogged(this) {
        putExtra(Constants.INTENT_ID, reminder.id)
      }
      finish()
    }
    alert.setPositiveButton(R.string.ok) { dialog, _ ->
      dialog.dismiss()
      finish()
    }
    val alertDialog = alert.create()
    alertDialog.setOnCancelListener { finish() }
    alertDialog.setOnDismissListener { finish() }
    alertDialog.show()

    Dialogues.setFullWidthDialog(alertDialog, this)
  }
}
